File: 剑指Offer/42.连续子数组的最大和.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maxSubSum = float('-inf')
        curSum = 0
        for num in nums:
            curSum = max(curSum + num, num)
            maxSubSum = max(maxSubSum, curSum)
        return maxSubSum

File: 剑指Offer/42.连续子数组的最大和.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int maxSubsum = INT_MIN;
        int curSum = 0;
        for (int i : nums) {
            curSum = max(curSum + i, i);
            maxSubsum = max(maxSubsum, curSum);
        }
        return maxSubsum;
    }
};